### PR TITLE
Uxui competitive analysis

### LINKED ISSUE #3888: [Term Entry] UI and UX Design UIUX (User Research) Competitive Analysis
### Reason/inspiration (optional)

We would like a new term entry on the `competitive-analysis` term for [UIUX](https://www.codecademy.com/resources/docs/uiux): Competitive Analysis. The entry should go in a new file under `docs/content/uiux/concepts/competitive-analysis/competitive-analysis.md`.

The entry should describe the term and how it is used. Be sure to include images to aid understanding, if required.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by christaton**
### Description

Creating docs entry for 'Competitive Analysis' in the Codecademy UI/UX content section. Includes an overview of the concept and the basic process.

Closes #3888 

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@christaton This is a great entry.

I noticed you have used personal pronouns like `you` all throughout the entry. You need to rephrase the sentences to remove any personal pronouns as we refrain from using them in entries. Please let me know once you are done making this change.

**Comment by christaton** [comment]
@ishg-153 Thank you for the feedback. I have updated and committed the changes, and re-requested review.

File: content/uiux/concepts/competitive-analysis/competitive-analysis.md
Comment (ishg-153) [inline content/uiux/concepts/competitive-analysis/competitive-analysis.md:34]:
```suggestion
Successful elements from competitors' interfaces can be integrated into the product while addressing their shortcomings to create a unique and improved user experience for the audience.
```

File: content/uiux/concepts/competitive-analysis/competitive-analysis.md
Comment (ishg-153) [inline content/uiux/concepts/competitive-analysis/competitive-analysis.md:14]:
```suggestion
In UI/UX design, **Competitive Analysis** is a crucial process for evaluating and benchmarking a digital product against competitors. By examining the user experience, [visual design](https://www.codecademy.com/resources/docs/uiux/visual-design), interaction elements, content strategy, and technological implementations of rival platforms, valuable insights are gained to inform and enhance design decisions.
```

File: content/uiux/concepts/competitive-analysis/competitive-analysis.md
Comment (ishg-153) [inline content/uiux/concepts/competitive-analysis/competitive-analysis.md:18]:
```suggestion
Understanding the competitive landscape helps to align with industry standards, identify user expectations, and foster innovation. This strategic analysis ensures that the digital interface meets user needs.
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@christaton I've suggested a couple of more changes. Please make those and ping me once done.

**Review by christaton — COMMENTED** [review]
Thanks for the feedback @ishg-153. Changes accepted and implemented.

**Review by ishg-153 — APPROVED** [review]
@christaton Great! This looks ready for a second review.

**Review by Helfese — APPROVED** [review]

**Comment by ishg-153** [comment]
Hey @Helfese. Only the maintainers (people with the collaborator tag) review and approve requests. I request you to not request or approve changes.

**Review by dakshdeepHERE — APPROVED** [review]
LGTM for merge 🚀

**Pull request merged**
